bin/build.sh

```diff
@@ -5,10 +5,189 @@
 # PORT_ROOT must be defined to the root directory of the cloned ZOSOpenTools port 
 # Either PORT_TARBALL or PORT_GIT must be defined (but not both). This indicates where to pull source from
 # 
-# A certificate of the form: tarball.pem or git.pem must be in the PORT_ROOT directory 
 # Each dependent tool will have it's corresponding environment set up by sourcing .env from the installation
 # directory. The .env will be searched for in $HOME/zot/prod/<tool>, /usr/zot/<tool>, $HOME/zot/boot/<tool>
 
+checkdeps() {
+	deps=$*
+	fail=false
+	for dep in $deps; do
+		if ! [ -f "${HOME}/zot/prod/${dep}/.env" ] && ! [ -f "${HOME}/zot/boot/${dep}/.env" ] && ! [ -f "/usr/zot/${dep}/.env" ] ; then
+			echo "Unable to find .env for dependency ${dep}" >&2
+			fail=true
+		fi
+	done
+	if $fail ; then
+		exit 4
+	fi
+}
+
+setdepsenv() {
+	for dep in $deps; do
+		if [ -f "${HOME}/zot/prod/${dep}/.env" ]; then
+			cd "${HOME}/zot/prod/${dep}"
+		elif [ -f "/usr/zot/${dep}/.env" ] ; then
+			cd "/usr/zot/${dep}"
+		elif [ -f "${HOME}/zot/boot/${dep}/.env" ]; then
+			cd "${HOME}/zot/boot/${dep}"
+		else 
+			echo "Internal error. Unable to find .env but earlier check should have caught this" >&2
+			exit 16
+		fi
+		. ./.env
+	done
+}
+
+gitclone() {
+	if ! git --version >$STDOUT 2>$STDERR ; then
+		echo "git is required to download from the git repo" >&2
+		exit 4
+	fi
+	gitname=$(basename $PORT_GIT_URL)
+	dir=${gitname%%.*}
+	if [ -d "${dir}" ]; then
+		echo "Using existing git clone'd directory ${dir}" >$STDERR
+	else
+		echo "Clone and create ${dir}" >$STDERR
+		if ! git clone "${PORT_GIT_URL}" 2>$STDERR; then
+                        echo "Unable to clone ${gitname} from ${PORT_GIT_URL}" >&2
+                        exit 4
+		fi
+		chtag -R -h -tcISO8859-1 "${dir}"
```

Comment:
Is this intended for git 2.14 or the later version?  Maybe it should apply to untagged files rather than already tagged files?  Although unlikely, there could be a case where a .gitattributes file is added to intentionally tag a file other than ASCII.

Reply:
yes - this should be smarter. also - I have seen cases now where .png and .pdf files are in the repo - and those should be skipped.